sample_dataset/metabase/sample_dataset/generate.clj

```diff
@@ -248,6 +271,43 @@
       person
       (assoc person :orders (vec (repeatedly num-orders #(random-order person (rand-nth products))))))))
 
+(defn- add-autocorrelation
+  ([k xs] (add-autocorrelation 1 k xs))
+  ([lag k xs]
+   (map (fn [prev next]
+          (update prev k #(/ (+ % (k next)) 2)))
+        xs
+        (drop lag xs))))
+
+(defn add-increasing-variance
+  [k xs]
+  (let [n (count xs)]
+    (map-indexed (fn [i x]
+                   (update x k * (+ 1 (* (/ i n) (- (rand 2) 0.9)))))
+                 xs)))
+
+(defn add-seasonality
```

Comment:
These sorts of things would benefit from having a short little blurb too about what exactly it is they do. One day someone else is going to have to go in and make changes to this code and we will be making their lives a lot easier if we make it easy for them to understand what everything does

Reply:
+1